### PR TITLE
Adding Fractal to project for styleguide generation

### PR DESCRIPTION
**Pull request by gestchild**

### CODE REVIEW

File: server/fractal.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+/* Set view templates file extension */
+fractal.components.set('ext', '.njk');
+
+/* Set preview template for components */
+fractal.components.set('default.preview', '@preview');
+
+const mandelbrot = require('@frctl/mandelbrot'); // require the Mandelbrot theme module
+
+// create a new instance with custom config options
+const myCustomisedTheme = mandelbrot({
```
Comment (davidpmccormick) [inline server/fractal.js]: `cardiganTheme`?

File: server/fractal.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+'use strict';
```
Comment (jamesgorrie) [inline server/fractal.js]: We don't need this - everything is set to that by default.

File: server/fractal.js (hunk 1 above)
Comment (jamesgorrie) [inline server/fractal.js]: ~~cardigan~~  too many projects named their joke names through common usage.

File: server/package.json [hunk 3]
```diff
@@ -15,6 +15,8 @@
   "author": "wellcomecollection.dev@wellcome.ac.uk",
   "license": "MIT",
   "dependencies": {
+    "@frctl/fractal": "^1.0.8",
+    "@frctl/nunjucks": "https://github.com/gestchild/nunjucks/tarball/66b327d3c9ac24f97784332c09669336fe4c650a",
```
Comment (jamesgorrie) [inline server/package.json]: Why have we done it this way?

File: server/views/components/grid-placeholder/README.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+These are some notes about the grid-placeholder component
+
+- list item
+- list item
+
+[This is a link](https://www.wellcomecollection.org)
```
Comment (jamesgorrie) [inline server/views/components/grid-placeholder/README.md]: I can't really see the purpose of this readme?

File: server/views/components/sample/sample.njk [hunk 5]
```diff
@@ -1,6 +1,7 @@
 <div class="grid__cell">
   <div class="sample-component">
     <h2 class="sample-component__heading">Sample component</h2>
-    <p>And some sample text</p>
+    <p>{{ text }}</p>
+    {% include "views/components/grid-placeholder.njk" %}
   </div>
 </div>
```
Comment (jamesgorrie) [inline server/views/components/sample/sample.njk]: Could we not have the `sample` directory? `sample` would suffice and keeps things flat and uncomplicated.

File: server/views/docs/index.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
+---
+title: Cardigan
```
Comment (jamesgorrie) [inline server/views/docs/index.md]: As mentioned above, too dangerous.

File: server/views/docs/layout.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+### Medium
+
+- Number of columns: 8
+- Width of gutters: 24px
+- Width of margins: 42px
+
+### Large
+- Number of columns: 12
+- Width of gutters: 30px
+- Width of margins: 60px
```
Comment (jamesgorrie) [inline server/views/docs/layout.md]: It'd be nice to generate this from the SCSS. Let's have a look.

File: server/fractal.js (hunk 1 above)
Comment (davidpmccormick) [inline server/fractal.js]: Yep, I wasn't suggesting it as a joke – deadly serious.

File: server/package.json (hunk 3 above)
Comment (gestchild) [inline server/package.json]: Using a fork for now as the current nunjucks adapter for fractal doesn't allow full paths for includes, which is what we're using in the app. Hoping to get the change into the main repo then can switch back

File: server/views/components/sample/sample.njk (hunk 5 above)
Comment (gestchild) [inline server/views/components/sample/sample.njk]: Having each component in its own directory lets us add notes and context data to be displayed in the styleguide.

File: server/views/docs/layout.md (hunk 7 above)
Comment (gestchild) [inline server/views/docs/layout.md]: Yeah, that would be good

File: server/views/components/sample/sample.njk (hunk 5 above)
Comment (jamesgorrie) [inline server/views/components/sample/sample.njk]: So we have a readme per unit? This is a lot of overhead no?
Or are we happy with that being the required amount of overhead per component.

In a way I like it as we can add metrics etc to them too - potentially even real times ones.

Comment (gestchild) [inline server/views/components/sample/sample.njk]: They aren't necessarily required, but I think will be useful.

Hadn't thought of metrics, but that would be great

**Pull request merged**